recipes/renode-cli/helpers/renode_build_with_dotnet.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+set -uex
+
+framework_version=$1
+shift 1
+
+OUTPUT_DIRECTORY="${SRC_DIR}/output"
+
+CONFIGURATION="Release"
+BUILD_PLATFORM="Any CPU"
+HEADLESS=true
+NET=true
+TFM="net${framework_version}"
+HOST_ARCH="i386"
+# Common cmake flags
+CMAKE_COMMON=""
+
+PARAMS=(
+  # By default use CC as Compiler- and LinkerPath, and AR as ArPath
+  ${CC:+"p:CompilerPath=$CC"}
+  ${CC:+"p:LinkerPath=$CC"}
+  ${AR:+"p:ArPath=$AR"}
+  # But allow users to override it
+  "$@"
+)
+
+if $HEADLESS
+then
+    BUILD_TARGET=Headless
+    PARAMS+=(p:GUI_DISABLED=true)
+else
+    BUILD_TARGET=Mono
+fi
+
```

Comment:
Let's **not**  attempt to support alternate build options that are not needed for us to build this in conda-forge.  We should only support building a headless build that does not use Mono (which is now effectively dead anyway), and only support native packages here and not `noarch`.  The simpler we can make this recipe, the easier it will be for others to understand and help to maintain.

Reply:
Agreed